src/main/java/uk/gov/hmcts/reform/pdf/service/service/AuthorisationService.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.hmcts.reform.pdf.service.service;
+
+import feign.FeignException;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.pdf.service.client.S2sClient;
+import uk.gov.hmcts.reform.pdf.service.exception.AuthorisationException;
+
+@Service
+public class AuthorisationService {
+
+    public static final String SERVICE_AUTHORISATION_HEADER = "ServiceAuthorization";
+
+    private final S2sClient s2sClient;
+
+    public AuthorisationService(S2sClient s2sClient) {
+        this.s2sClient = s2sClient;
+    }
+
+    /**
+     * Authenticates the service.
+     * Returns service name on success.
+     */
+    public String authorise(String serviceAuthHeader) {
+        try {
+            return s2sClient.getServiceName(serviceAuthHeader);
+        } catch (FeignException exc) {
+            boolean isClientError = exc.status() >= 400 && exc.status() <= 499;
```

Comment:
Are all 4XX responses authentication errors?

Reply:
same - reusing. 4xx stands for client errors and can be rolled out as unauthorised regardless the origin. 5xx are more important hence exception handler is responding back with exact same code

Reply:
I guess it makes sense that if the client does something wrong when trying to authorise it will result in him being not authorised. You're right, I'm fine with that 👍